src/core/tests/ov_tensor_test.cpp

```diff
@@ -51,22 +51,37 @@ TEST_F(OVTensorTest, operators) {
     ASSERT_TRUE(!t);
 }
 
-class OVMockAllocator : public ov::AllocatorImpl {
-public:
-    MOCK_METHOD(void*, allocate, (size_t, size_t), ());
-    MOCK_METHOD(void, deallocate, (void*, size_t, size_t), ());                  // NOLINT(readability/casting)
-    MOCK_METHOD(bool, is_equal, (const ov::AllocatorImpl&), (const, noexcept));  // NOLINT(readability/casting)
+struct OVMockAllocator {
```

Comment:
should we keep old 'OVMockAllocator' test to ensure that 'deprecated' API still works for end user?

Reply:
added